src/main/java/org/folio/bulkops/controller/BulkOperationController.java

```diff
@@ -256,4 +262,30 @@ public ResponseEntity<List<String>> getListUsedTenants(UUID operationId) {
   public ResponseEntity<BulkOperationDto> triggerBulkEditByQuery(UUID xOkapiUserId, QueryRequest queryRequest) {
     return new ResponseEntity<>(bulkOperationMapper.mapToDto(bulkOperationService.triggerByQuery(xOkapiUserId, queryRequest)), HttpStatus.OK);
   }
+
+
+  @Override
+  public ResponseEntity<ProfileSummaryResultsDto> getProfiles(String query, Integer offset, Integer limit) {
+    var response = profileService.getProfileSummaries(query, offset, limit);
+    return ResponseEntity.ok(response);
+  }
+
+
+  @Override
+  public ResponseEntity<ProfileDto> createProfile(ProfileRequest profileRequest) {
```

Comment:
@kjain110  , desired permission "bulk-operations.profiles.item.lock.execute" must be explicitly checked to allow or restrict creating of profile with locked=true.